dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/CategoryOptionComboObjectBundleHook.java

```diff
@@ -0,0 +1,141 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.dxf2.metadata.objectbundle.hooks;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import org.hisp.dhis.category.CategoryCombo;
+import org.hisp.dhis.category.CategoryOption;
+import org.hisp.dhis.category.CategoryOptionCombo;
+import org.hisp.dhis.category.CategoryService;
+import org.hisp.dhis.dxf2.metadata.objectbundle.ObjectBundle;
+import org.hisp.dhis.feedback.ErrorCode;
+import org.hisp.dhis.feedback.ErrorReport;
+import org.springframework.stereotype.Component;
+
+@Component
+@AllArgsConstructor
+public class CategoryOptionComboObjectBundleHook
+    extends AbstractObjectBundleHook<CategoryOptionCombo> {
+  private final CategoryService categoryService;
+
+  static boolean haveEqualCatComboCatOptionReferenceIds(
+      CategoryOptionCombo one, CategoryOptionCombo other) {
+    if (one == null || other == null) {
+      return false;
+    }
+
+    if (one.getCategoryCombo() == null || other.getCategoryCombo() == null) {
+      return false;
+    }
+
+    if (one.getCategoryOptions() == null || other.getCategoryOptions() == null) {
+      return false;
+    }
+
+    if (!one.getCategoryCombo().getUid().equals(other.getCategoryCombo().getUid())) {
+      return false;
+    }
+
+    Set<String> oneCategoryOptionUids =
+        one.getCategoryOptions().stream().map(CategoryOption::getUid).collect(Collectors.toSet());
+
+    Set<String> otherCategoryOptionUids =
+        other.getCategoryOptions().stream().map(CategoryOption::getUid).collect(Collectors.toSet());
+
+    return oneCategoryOptionUids.equals(otherCategoryOptionUids);
+  }
+
+  private void checkDuplicateCategoryOptionCombos(
+      CategoryOptionCombo categoryOptionCombo,
+      ObjectBundle bundle,
+      Consumer<ErrorReport> addReports) {
+
+    if (bundle.isPersisted(categoryOptionCombo)) {
+      return; // Only check for duplicates if the object is not persisted
+    }
+
+    List<CategoryOptionCombo> categoryOptionCombos =
+        categoryService.getAllCategoryOptionCombos().stream()
+            .filter(
+                coc ->
+                    coc.getCategoryCombo()
+                        .getUid()
+                        .equals(categoryOptionCombo.getCategoryCombo().getUid()))
+            .toList();
```

Comment:
This looks like it should use a dedicated method only loading COC with the tagret CC ID instead to avoid loading a big number of DB objects into memory. 

Also I wonder if this logic is correct. Don't you want to check if any other CC has the same categories but is not the same CC?

Reply:
Resolving this comment as this method has been removed from the PR